src/providers/azure.rs

```diff
@@ -0,0 +1,519 @@
+//! Azure backend for tsunami.
+//!
+//! The primary `impl Launcher` type is [`Launcher`].
+//! It internally uses the lower-level, region-specific [`RegionLauncher`].
+//! Both these types use [`Setup`] as their descriptor type.
+//!
+//! Azure does not support Spot or Defined Duration instances.
+//! As a result, if your tsunami crashes (i.e., exits without calling `drop()` on [`Launcher`], you must manually terminate your instances
+//! to avoid extra costs.
+//! The easiest way to do this is to delete resource groups beginning with `tsunami_`:
+//! `az group delete --name <name> --yes`.
+//! You can find such resource groups using:
+//! `az group list`.
+//!
+//! To use this provider, you must [install the Azure
+//! CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest),
+//! and the `az` command must be in your `$PATH`.
+//! Also, you must run `az login` to authenticate with Microsoft.
+//!
+//! # Example
+//! ```rust,no_run
+//! use tsunami::providers::{azure, Launcher};
+//! use tsunami::TsunamiBuilder;
+//!
+//! let mut b = TsunamiBuilder::default();
```

Comment:
Here too I'd like to see a more elaborate example.

Reply:
What do you think the example should do?

Reply:
See, for example, the old main example:
https://github.com/jonhoo/tsunami/blob/96a2f912f445b57b02a217b4a6998837373989b6/src/lib.rs#L7-L38

Reply:
what about: 
```
//! use tsunami::TsunamiBuilder;
//! use tsunami::providers::{Launcher, aws};
//! use rusoto_core::{DefaultCredentialsProvider, Region};
//! fn main() -> Result<(), failure::Error> {
//!     // Initialize AWS
//!     let mut aws = aws::Launcher::default();
//!
//!     // Initialize a TsunamiBuilder
//!     let mut tb = TsunamiBuilder::default();
//!     tb.use_term_logger();
//!
//!     // Create a machine descriptor and add it to the Tsunami
//!     let m = aws::Setup::default()
//!         .region_with_ubuntu_ami(Region::UsWest1) // default was UsEast1
//!         .setup(|ssh| { // default is a no-op
//!             ssh.run("sudo apt update")?;
//!             ssh.run("curl https://sh.rustup.rs -sSf | sh -- -y")?;
//!         });
//!     tb.add("my_vm", m);
//!
//!     // Launch the VM
//!     tb.spawn(&mut aws)?;
//!
//!     // SSH to the VM and run a command on it
//!     let vms = aws.connect_all()?;
//!     let my_vm = vms.get("my_vm").unwrap();
//!     println!("public ip: {}", my_vm.public_ip);
//!     let ssh = my_vm.ssh.as_ref().unwrap();
//!     ssh.cmd("git clone https://github.com/jonhoo/tsunami")?;
//!     ssh.cmd("cd tsunami && cargo build")?;
//!     Ok(())
//! }
```

Reply:
I like it!